packages/react-router-dom/index.tsx

```diff
@@ -405,9 +405,12 @@ export function useSearchParams(defaultInit?: URLSearchParamsInit) {
       nextInit: URLSearchParamsInit,
       navigateOptions?: { replace?: boolean; state?: any }
     ) => {
-      navigate("?" + createSearchParams(nextInit), navigateOptions);
+      navigate({
+        hash: location.hash,
+        search: "?" + createSearchParams(nextInit),
+      }, navigateOptions);
     },
-    [navigate]
+    [location.hash, navigate]
```

Comment:
This seems like the wrong place to fix it. This is only addressing a very specific error condition. This should be fixed inside of `navigate` instead so it doesn't overwrite non-search portions of the current location when navigating.

Reply:
Makes sense @timdorr, will try to fix it in `navigate` then

Reply:
@timdorr I've updated the fix implementation on `useNavigate` hook

Reply:
Let me know if you want to keep the history of the previous fix in this conversation or if you want me to clean up the git history